### PR TITLE
[MI-2119]: Jira's autolink should issue links that contain a comment …

### PR DESCRIPTION
**Pull request by Nityanand13**
Issue link:- [#773](https://github.com/mattermost/mattermost-plugin-jira/issues/773)
PR link:- [#839](https://github.com/mattermost/mattermost-plugin-jira/pull/839)
Internal Ticket link:- [MI-2172](https://brightscout.atlassian.net/browse/MI-2172)

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -356,32 +357,38 @@ func (p *Plugin) AddAutolinksForCloudInstance(ci *cloudInstance) error {
 		return fmt.Errorf("unable to get project keys: %w", err)
 	}
 
-	for _, proj := range plist {
-		key := proj.Key
-		err = p.AddAutolinks(key, ci.BaseURL)
-	}
+	err = p.AddAutolinks(plist, ci.BaseURL)
```
Comment (Kshitij-Katiyar) [inline server/plugin.go:360]:
```suggestion
	if err = p.AddAutolinks(plist, ci.BaseURL); err!= nil {
```

File: server/plugin.go [hunk 2]
```diff
@@ -356,32 +357,38 @@ func (p *Plugin) AddAutolinksForCloudInstance(ci *cloudInstance) error {
...
 	if err != nil {
 		return fmt.Errorf("some keys were not installed: %w", err)
 	}
 
 	return nil
 }
 
-func (p *Plugin) AddAutolinks(key, baseURL string) error {
+func (p *Plugin) AddAutolinks(plist jira.ProjectList, baseURL string) error {
```
Comment (Kshitij-Katiyar) [inline server/plugin.go:368]:
```suggestion
func (p *Plugin) AddAutolinks(projectList jira.ProjectList, baseURL string) error {
```

File: server/plugin.go [hunk 3]
```diff
@@ -356,32 +357,38 @@ func (p *Plugin) AddAutolinksForCloudInstance(ci *cloudInstance) error {
...
-			Name:     key + " link to key for " + baseURL,
-			Pattern:  `(` + strings.ReplaceAll(baseURL, ".", `\.`) + `/browse/)(` + key + `)(-)(?P<jira_id>\d+)`,
-			Template: `[` + key + `-${jira_id}](` + baseURL + `/browse/` + key + `-${jira_id})`,
+			Name:     "Link to key for " + baseURL,
+			Pattern:  `(` + strings.ReplaceAll(baseURL, ".", `\.`) + `/browse/)(?P<project_id>\w+)(-)(?P<jira_id>\d+)`,
+			Template: `[${project_id}-${jira_id}](` + baseURL + `/browse/${project_id}-${jira_id})`,
 		},
 	}
 
+	for _, proj := range plist {
```
Comment (Kshitij-Katiyar) [inline server/plugin.go:383]:
```suggestion
	for _, project := range plist {
```

**Review by Kshitij-Katiyar — APPROVED** [review]

**Comment by manojmalik20** [comment]
Please add the issue link, ticket link and the original PR link in the PR description.

**Review by manojmalik20 — APPROVED** [review]
Make sure you've tested this thoroughly.

**Pull request merged**